src/ensembl/src/shared/components/in-app-search/InAppSearch.tsx

```diff
@@ -14,54 +14,132 @@
  * limitations under the License.
  */
 
-import React, { useState } from 'react';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import classNames from 'classnames';
+import upperFirst from 'lodash/upperFirst';
 
-import { isEnvironment, Environment } from 'src/shared/helpers/environment';
+import {
+  search,
+  clearSearch,
+  updateQuery
+} from 'src/shared/state/in-app-search/inAppSearchSlice';
+import {
+  getSearchQuery,
+  getSearchResults
+} from 'src/shared/state/in-app-search/inAppSearchSelectors';
+
+import { pluralise } from 'src/shared/helpers/formatters/pluralisationFormatter';
+import { getCommaSeparatedNumber } from 'src/shared/helpers/formatters/numberFormatter';
 
 import SearchField from 'src/shared/components/search-field/SearchField';
+import { PrimaryButton } from 'src/shared/components/button/Button';
 import QuestionButton, {
   QuestionButtonOption
 } from 'src/shared/components/question-button/QuestionButton';
+import CloseButton from 'src/shared/components/close-button/CloseButton';
+import InAppSearchMatches from './InAppSearchMatches';
+
+import type { RootState } from 'src/store';
+import type { AppName } from 'src/shared/state/in-app-search/inAppSearchSlice';
 
 import styles from './InAppSearch.scss';
 
-type Props = {
-  className?: string;
+export type InAppSearchMode = 'interstitial' | 'sidebar';
+
+export type Props = {
+  app: AppName;
+  genomeId: string;
+  mode: InAppSearchMode;
 };
 
 const InAppSearch = (props: Props) => {
-  const [value, setValue] = useState('');
-
-  const searchFieldWrapperClasses = classNames(
-    styles.searchFieldWrapper,
-    props.className
+  const { app, genomeId, mode } = props;
+  const query = useSelector((state: RootState) =>
+    getSearchQuery(state, props.app, props.genomeId)
```

Comment:
Why `props.app` and `props.genomeId` when they've been destructured in line 57?

Reply:
✅